doc/post-transaction-actions.rst

```diff
@@ -0,0 +1,89 @@
+  ANY WARRANTY expressed or implied, including the implied warranties of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+  Public License for more details.  You should have received a copy of the
+  GNU General Public License along with this program; if not, write to the
+  Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+  02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+  source code or documentation are not subject to the GNU General Public
+  License and may only be used or replicated with the express permission of
+  Red Hat, Inc.
+
+===================================
+DNF post-transaction-actions Plugin
+===================================
+
+-----------
+Description
+-----------
+
+The plugin allows to define actions to be executed upon completing an RPM transaction. Each action
+may define a (glob-like) filtering rule on the package NEVRA or package files, as well as whether
+the package was installed or removed. Actions are defined in action files.
+
+-------------
+Configuration
+-------------
+
+The plugin configuration is in ``/etc/dnf/plugins/post-transaction-actions.conf``. All configuration
+options are in the ``[main]`` section.
+
+``enabled``
+    Whether the plugin is enabled. Default value is ``True``.
+
+``actiondir``
+    Path to the directory with action files. Action files must have the ".action" extension.
+    Default value is "/etc/dnf/plugins/post-transaction-actions.d/".
+
+------------------
+Action file format
+------------------
+
+Empty lines and lines that start with a '#' character are ignored.
+Each non-comment line defines an action and consists of three items separated by colons:
+package_filter, transaction_state, command.
+
+``package_filter``
+   A (glob-like) filtering rule aplied on the package NEVRA (also in the shortened forms) or
+   package files.
+
+``transaction_state``
+   Filters packages according to their state in the transaction.
+
+   * ``forward`` - packages that appeared on the system (includes reinstall packages)
+   * ``backward`` - packages that got removed from the system
+   * ``any`` - all packages
+
+``command``
+   Any shell command. Commands can contain variables.
+
+   The same shell command (after variables substitution) is executed only once per transaction.
+   The order of execution of the commands may differ from the order of the packages in the transaction.
+
+   The following variables are supported:
+      * ``${name}``, ``$name`` - package name
+      * ``${arch}``, ``$arch`` - package arch
+      * ``${ver}``, ``$ver`` - package version
+      * ``${rel}``, ``$rel`` - package release
+      * ``${epoch}``, ``$epoch`` - package epoch
+      * ``${repoid}``, ``$repoid`` - package repository id
+      * ``${state}``, ``$state`` - "downgrade", "downgraded", "install", "obsolete", "obsoleted",
+                                   "reinstall", "reinstalled", "remove", "upgrade", "upgraded"
```

Comment:
So for this part, I'd first do:
> Any shell command. The following variables in the command will be substituted:
>    (list of variables)

And then in a next paragraph, I'd describe the actual behavior a bit more thoroughly, my take:
> The shell command will be evaluated for each package that matched the `package_filter`. However, after variable substitution, any duplicate commands will be removed and each command will only be executed once per transaction. The order of execution of the commands may differ from the order of packages in the transaction.